scripts/jenkins/qa_openstack.sh

```diff
@@ -321,8 +321,14 @@ esac
 
 glance image-list
 imgid=$(glance image-list|grep debian-5|cut -f2 -d" ")
-mkdir -p ~/.ssh
-( umask 77 ; nova keypair-add testkey > ~/.ssh/id_rsa )
+
+if [ -f /root/.ssh/id_rsa.pub ]; then
+    nova keypair-add --pub-key /root/.ssh/id_rsa.pub testkey
+else
+    umask 077
```

Comment:
this changes the umask globally rather than in a subshell, like before. I don't think thats a good idea. Please revert it back to the subshell handling that it was before.